dev/com.ibm.ws.microprofile.health.2.0_fat/fat/src/com/ibm/ws/microprofile/health20/fat/DelayAppStartupHealthCheckTest.java

```diff
@@ -73,6 +75,69 @@ public void cleanUp() throws Exception {
         server1.stopServer("CWWKE1102W", "CWWKE1105W", "CWMH0052W", "CWMH0053W");
     }
 
+    @Test
+    public void testReadinessEndpointOnServerStart() throws Exception {
+        server1.setMarkToEndOfLog();
+        server1.stopServer("CWWKE1102W", "CWWKE1105W", "CWMH0052W", "CWMH0053W");
+
+        class StartServerOnThread extends Thread {
+            @Override
+            public void run() {
+                try {
+                    server1.startServer();
+                } catch (Exception e) {
+                    assertTrue("Failure to start server on a seperate thread.", server1.isStarted());
+                }
+            }
+        }
+
+        // Need to ensure the server is not finish starting when readiness endpoint is hit so start the server on a separate thread
+        // Note: this does not guarantee that we hit the endpoint during server startup, but it is highly likely that it will
+        StartServerOnThread startServerThread = new StartServerOnThread();
+        log("testReadinessEndpointOnServerStart", "Starting DelayedHealthCheck server on separate thread.");
+        startServerThread.start();
+
+        // Hit the readiness endpoint as the server is still starting up
+        // We expect a connection refused as the ports are not open until server is fully started
+        try {
+            log("testReadinessEndpointOnServerStart", "Testing the /health/ready endpoint as the server is still starting up.");
+            HttpURLConnection conReady = null;
+            String responseCode = null;
+            boolean connectionExceptionEncountered = false;
+
+            try {
+                conReady = HttpUtils.getHttpConnectionWithAnyResponseCode(server1, READY_ENDPOINT);
```

Comment:
we are only pinging the `/health/ready` endpoint once here, we should put this inside a loop and repeatedly ping the `/health/ready` endpoint until it returns HTTP 200